apps/dh/api-dh/source/DataHub.WebApi/Clients/Wholesale/ProcessManager/NotifyAggregatedMeasureDataClientAdapter.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.ProcessManager.Client.Processes.BRS_023_027.V1;
+using Energinet.DataHub.WebApi.Clients.Wholesale.v3;
+using Energinet.DataHub.WebApi.GraphQL.Enums;
+using Energinet.DataHub.WebApi.GraphQL.Types.Calculation;
+
+namespace Energinet.DataHub.WebApi.Clients.Wholesale.ProcessManager;
+
+/// <inheritdoc/>
+internal class NotifyAggregatedMeasureDataClientAdapter(
+    INotifyAggregatedMeasureDataClientV1 innerClient)
+        : INotifyAggregatedMeasureDataClientAdapter
+{
+    private readonly INotifyAggregatedMeasureDataClientV1 _innerClient = innerClient;
+
+    /// <inheritdoc/>
+    public async Task<IEnumerable<CalculationDto>> QueryCalculationsAsync(
+        CalculationQueryInput input,
+        CancellationToken calculationToken = default)
+    {
+        // TODO:
+        // We currently only support filtering on one state initially, and only
+        // top-level states (orchestration instance states, not steps)
+        var states = input.States ?? [];
+        var state = states
+            .Select(x => x.MapToLifecycleState())
+            .FirstOrDefault();
```

Comment:
question: is the plan to then support multiple states in the `SearchCalculationsAsync`? 🙂

Reply:
Okay thanks! Should we implement it "correctly" here at this time then? The way I understand this code, is that if you filter for multiple states, you would actually only get results for the first state. Wouldn't it be more correct to search for all states and then filter the result instead (as is already happening here: https://github.com/Energinet-DataHub/greenforce-frontend/pull/3708/files#diff-f27e9c5d48c523aceeac4ccbf4c80aa206ed6afa5a065f9dfadf573923fd4061R70)

Reply:
Nice catch, I will change the filter values to `null` and write a TODO for later.